qiskit/algorithms/eigen_solvers/eigen_solver.py

```diff
@@ -18,18 +18,34 @@
 import numpy as np
 
 from qiskit.opflow import OperatorBase
+from qiskit.utils.deprecation import deprecate_function
 from ..algorithm_result import AlgorithmResult
 from ..list_or_dict import ListOrDict
 
 
 class Eigensolver(ABC):
-    """The Eigensolver Interface.
+    """Pending deprecation: Eigensolver Interface.
+
+    The Eigensolver interface has been superseded by the
+    qiskit.algorithms.eigensolvers.Eigensolver interface.
```

Comment:
In the docstrings here, this being one example, we should be able to do ``:class:`qiskit.algorithms.eigensolvers.Eigensolver` `` right, which will be highlighted at present and when the replacement is available should give a clickable link in the docs.